plugin/src/Internal/Cache.hs

```diff
@@ -357,74 +454,166 @@ updateCache target InterpCache {..} newLoaderState newSymbols newNames cache = d
+    kvs = sortBy (comparing (length . fst)) (first moduleNameString <$> Map.toList m)
+
 report ::
   MonadIO m =>
   MVar Log ->
-  MVar Cache ->
+  HscEnv ->
   Target ->
+  Cache ->
   m ()
-report logVar cacheVar target =
-  liftIO (readMVar cacheVar) >>= \ Cache {stats} ->
-    for_ (Map.lookup target stats) \ CacheStats {restore, update} -> do
-      let
-        restoreStats =
-          text (show (length restore.loader.newBcos)) <+> text "BCOs" $$
-          text (show restore.loader.linker.newClosures) <+> text "closures" $$
-          text (show restore.symbols.new) <+> text "symbols" $$
-          text (show restore.loader.sameBcos) <+> text "BCOs already in cache"
-
-        newBcos = text <$> update.loader.newBcos
-
-        updateStats =
-          (if null newBcos then text "No new BCOs" else text "New BCOs:" <+> fsep (punctuate comma newBcos)) $$
-          text (show update.loader.linker.newClosures) <+> text "new closures" $$
-          text (show update.symbols.new) <+> text "new symbols" $$
-          text (show update.loader.sameBcos) <+> text "BCOs already in cache"
-
-      logd logVar $ hang (text target.get Outputable.<> text ":") 2 $
-        hang (text "Restore:") 2 restoreStats $$
-        hang (text "Update:") 2 updateStats
-
-withHscState :: (MVar OrigNameCache -> MVar (Maybe LoaderState) -> MVar SymbolMap -> IO ()) -> Ghc ()
-withHscState use =
-  withSession \ HscEnv {hsc_interp, hsc_NC = NameCache {nsNames}} ->
-#if __GLASGOW_HASKELL__ >= 910
-    for_ hsc_interp \ Interp {interpLoader = Loader {loader_state}, interpLookupSymbolCache} ->
-      liftIO $ use nsNames loader_state interpLookupSymbolCache
+report logVar _ target Cache {stats} =
+  for_ (stats !? target) \ CacheStats {restore, update, finder} -> do
+    let
+      restoreStats =
+        text (show (length restore.loader.newBcos)) <+> text "BCOs" $$
+        text (show restore.loader.linker.newClosures) <+> text "closures" $$
+        text (show restore.symbols.new) <+> text "symbols" $$
+        text (show restore.loader.sameBcos) <+> text "BCOs already in cache"
+
+      newBcos = text <$> update.loader.newBcos
+
+      updateStats =
+        (if null newBcos then text "No new BCOs" else text "New BCOs:" <+> fsep (punctuate comma newBcos)) $$
+        text (show update.loader.linker.newClosures) <+> text "new closures" $$
+        text (show update.symbols.new) <+> text "new symbols" $$
+        text (show update.loader.sameBcos) <+> text "BCOs already in cache"
+
+      finderStats =
+        hang (text "Hits:") 2 (moduleColumns finder.hits) $$
+        hang (text "Misses:") 2 (moduleColumns finder.misses)
+
+    logd logVar $ hang (text target.get Outputable.<> text ":") 2 $
+      hang (text "Restore:") 2 restoreStats $$
+      hang (text "Update:") 2 updateStats $$
+      hang (text "Finder:") 2 finderStats
+
+#if __GLASGOW_HASKELL__ >= 911
+
+newFinderCache :: MVar Cache -> Cache -> Target -> IO FinderCache
```

Comment:
Could you highlight how this differs from the current GHC FinderCache implementation? (From the application Cache state). As a note. Thanks.